File: app/src/main/java/com/paradoxo/amadeus/dao/BDGateway.java
package com.paradoxo.amadeus.dao;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;

public class BDGateway {

    private static BDGateway bdGateway;
    private SQLiteDatabase db;

    public BDGateway(Context ctx) {
        BDHelper helper = new BDHelper(ctx);
        db = helper.getWritableDatabase();
    }

    public static BDGateway getInstance(Context ctx) {
        if (bdGateway == null)
            bdGateway = new BDGateway(ctx);
        return bdGateway;
    }

    public SQLiteDatabase getDatabase() {
        return this.db;
    }
}

File: app/src/main/java/com/paradoxo/amadeus/dao/BDGateway.kt
package com.paradoxo.amadeus.dao

import android.content.Context
import android.database.sqlite.SQLiteDatabase

class BDGateway private constructor(ctx: Context) {
    val database: SQLiteDatabase = BDHelper(ctx).writableDatabase

    companion object {
        private var instance: BDGateway? = null

        fun getInstance(ctx: Context): BDGateway {
            if (instance == null) instance = BDGateway(ctx)
            return instance!!
        }
    }
}
